deploy-board/deploy_board/static/js/components/capacitycomponents.js

```diff
@@ -313,8 +327,35 @@ Vue.component("static-capacity-config", {
                     this.sizeWarning = getCapacityScaleDownAlertMessage(true, true);
                 }
             }
+
             this.imbalanceWarning = checkImbalance(this.placements, calculateImbalanceThreshold(sizeIncrease, this.placements.length));
             this.showImbalanceWarning = this.imbalanceWarning != '';
+
+            if (this.terminationLimit !== null) {
+                let terminatingHostCount = 0;
+                $.ajax({
+                    type: 'GET',
+                    url: `/groups/${this.groupName}/hosts`,
+                    dataType: "json",
+                    beforeSend: function(xhr, settings) {
+                        var csrftoken = getCookie('csrftoken');
+                        xhr.setRequestHeader("X-CSRFToken", csrftoken);
+                    },
+                    success: function (data) {
+                        terminatingHostCount = data.length;
+                    },
+                    error: function () {
+                        terminatingHostCount = 0;
+                    }
+                });
+                if (-sizeIncrease > this.terminationLimit) {
+                    this.showTerminationError = true;
+                    this.terminationError = getTerminationLimitAlertMessage(false);
+                } else if (-sizeIncrease > this.terminationLimit - terminatingHostCount) {
+                    this.showTerminationError = true;
+                    this.terminationError = getTerminationLimitAlertMessage(true);
```

Comment:
One bug in this piece of code is that you are using an asynchronous ajax call to get the `terminatingHostCount` value, but you are using it in a synchronous way to check the `terminationLimit` condition. This means that your code may not wait for the ajax call to finish before proceeding to the next statements, which could lead to unexpected results.

According to some web sources[1](https://stackoverflow.com/questions/13971769/how-to-make-jquery-ajax-request-synchronous)[2](https://stackoverflow.com/questions/6685249/jquery-performing-synchronous-ajax-requests), you can make your ajax call synchronous by setting the async option to false. [However, this is not recommended as it can cause your browser to become unresponsive](https://varunon9.medium.com/what-can-go-wrong-if-you-make-synchronous-ajax-calls-using-jquery-understanding-event-loop-in-9b597c7a87a1)[3](https://varunon9.medium.com/what-can-go-wrong-if-you-make-synchronous-ajax-calls-using-jquery-understanding-event-loop-in-9b597c7a87a1). A better way is to use a callback function or a promise object to handle the response of the ajax call and perform the necessary actions after it is completed.

Reply:
https://stackoverflow.com/questions/5316697/jquery-return-data-after-ajax-call-success

See the UPDATE (2017)